ratis-server/src/test/java/org/apache/ratis/server/impl/RaftReconfigurationBaseTest.java

```diff
@@ -339,17 +336,6 @@ void runTestBootstrapReconf(int numNewPeer, boolean startNewPeer, CLUSTER cluste
         final PeerChanges c1 = cluster.addNewPeers(numNewPeer, startNewPeer);
         LOG.info("Start changing the configuration: {}",
                 asList(c1.allPeersInNewConf));
-        final AtomicReference<Boolean> success = new AtomicReference<>();
-
-        Thread clientThread = new Thread(() -> {
-          try {
-            RaftClientReply reply = client.admin().setConfiguration(c1.allPeersInNewConf);
-            success.set(reply.isSuccess());
-          } catch (IOException ioe) {
-            LOG.error("FAILED", ioe);
-          }
-        });
-        clientThread.start();
```

Comment:
The client should be started before "startNewPeer" below.  Otherwise, the test is changed.

Reply:
I see, it took me some time to understand this case correctly, but I think I got it now. My most recent commit restores execution order, but at the same time also fixes concurrency problems.